### PR TITLE
Feat: Store details on store products page

### PR DESCRIPTION
**Pull request by CodeNinja500**
Store details and search box added.
Search has no functionality yet.

### CODE REVIEW

File: src/app/components/store-products/store-products.component.ts [hunk 1]
```diff
@@ -8,4 +19,39 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
...
+    return {
+      id: store.id,
+      name: store.name,
+      logoUrl: store.logoUrl,
+      distanceInKm: Math.round(store.distanceInMeters / 100) / 10,
+      tags: store.tagIds.map((tagId) => tagsMap[tagId]?.name)
+    };
+  }
+
+  onStoreSearchSubmitted(storeSearch: FormGroup): void {
```
Comment (gradzio) [inline src/app/components/store-products/store-products.component.ts]: we should probably remove it

File: src/app/components/store-products/store-products.component.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                            <span class="position-absolute end-0 top-0 mt-2 me-3" (click)="onStoreSearchSubmitted(storeSearch)">
+                                <svg xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 24 24" fill="none"
+                                    stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"
+                                    class="feather feather-search">
+                                    <circle cx="11" cy="11" r="8"></circle>
+                                    <line x1="21" y1="21" x2="16.65" y2="16.65"></line>
+                                </svg>
+                            </span>
+                            <!--for debugging-->
+                            <p>{{storeSearch.value |json}}</p>
```
Comment (gradzio) [inline src/app/components/store-products/store-products.component.html]: let's remove it :)

**Review by gradzio — CHANGES_REQUESTED** [review]

File: src/app/components/store-products/store-products.component.html (hunk 2 above)
Comment (CodeNinja500) [reply src/app/components/store-products/store-products.component.html]: Removed :)

File: src/app/components/store-products/store-products.component.ts (hunk 1 above)
Comment (CodeNinja500) [reply src/app/components/store-products/store-products.component.ts]: This function will be used later so I only removed the body that I used for debugging.

**Review by gradzio — APPROVED** [review]

**Pull request merged**